pkg/tcpip/network/ipv4/icmp.go

```diff
@@ -133,6 +133,43 @@ func (e *endpoint) handleICMP(r *stack.Route, netHeader buffer.View, vv buffer.V
 
 	case header.ICMPv4Timestamp:
 		received.Timestamp.Increment()
+		// Only send a reply if the checksum is valid.
+         	wantChecksum := h.Checksum()
+        	// Reset the checksum field to 0 to can calculate the proper
+        	// checksum. We'll have to reset this before we hand the packet off.
+        	h.SetChecksum(0)
+            	gotChecksum := ^header.ChecksumVV(vv, 0 /* initial */)
+            	if gotChecksum != wantChecksum {
+                	// It's possible that a raw socket expects to receive this.
+                	h.SetChecksum(wantChecksum)
+                	e.dispatcher.DeliverTransportPacket(r, header.ICMPv4ProtocolNumber, netHeader, vv)
+                	received.Invalid.Increment()
+           	 	return
+        	}
+        	// It's possible that a raw socket expects to receive this.
```

Comment:
These three lines are common between the If and otherwise. I think you should move them above the If so that you’ve got:

// It’s possible...
h.SetChecksum(...
e.dispatcher...
if gotChecksum != wantChecksum {
  received.Invalid...
  return
}